### PR TITLE
[robin-hood-hashing] Add package introducing robin-hood-hashing library

### PR DESCRIPTION
**Pull request by Talkless**
Specify library name and version:  **robin-hood-hashing/3.7.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/robin-hood-hashing/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import os
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/conanfile.py:3]:
```suggestion
import os
```

File: recipes/robin-hood-hashing/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+
+        self.copy(pattern="robin_hood.h", dst="include",
+                  src=os.path.join(self._source_subfolder, "src", "include"))
+
+    def package_info(self):
+        # Original CMakeLists.txt exports "robin_hood::robin_hood" target:
+        self.cpp_info.name = "robin_hood"
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/conanfile.py:33]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "robin_hood"
        self.cpp_info.names["cmake_find_package_multi"] = "robin_hood"
```

File: recipes/robin-hood-hashing/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+project(test_package CXX)
+cmake_minimum_required(VERSION 2.8.12)
+
+set(CMAKE_VERBOSE_MAKEFILE ON)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/test_package/CMakeLists.txt:6]:
```suggestion
cmake_minimum_required(VERSION 2.8.12)
project(test_package CXX)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
```

File: recipes/robin-hood-hashing/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+conan_basic_setup(TARGETS)
+
+find_package(robin_hood REQUIRED CONFIG)
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/test_package/CMakeLists.txt:9]:
```
# Testing correct name of robin-hood target
find_package(robin_hood REQUIRED CONFIG)
if(NOT TARGET robin_hood::robin_hood)
    message(FATAL_ERROR "robin_hood::robin_hood target not found")
endif()
```

File: recipes/robin-hood-hashing/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+add_executable(${CMAKE_PROJECT_NAME} test_package.cpp)
+target_link_libraries(${CMAKE_PROJECT_NAME} CONAN_PKG::robin_hood)
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/test_package/CMakeLists.txt:12]:
```suggestion
target_link_libraries(${CMAKE_PROJECT_NAME} ${CONAN_LIBS})
```

File: recipes/robin-hood-hashing/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake_find_package_multi", "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        self.run(os.path.join("bin","test_package"), run_environment=True)
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self.settings):
            self.run(os.path.join("bin","test_package"), run_environment=True)
```

File: recipes/robin-hood-hashing/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import os
+from conans import ConanFile, CMake, tools, RunEnvironment
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/test_package/conanfile.py:2]:
```suggestion
from conans import ConanFile, CMake, tools
```

File: recipes/robin-hood-hashing/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    generators = "cmake"
+
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/conanfile.py:24]: It seems you are settings c++14 in test package, is this correct?
```suggestion
        os.rename(extracted_dir, self._source_subfolder)

    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, 14)

    def package(self):
```

File: recipes/robin-hood-hashing/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+from conans import ConanFile, tools
+
+class ConanRecipe(ConanFile):
+    name = "robin-hood-hashing"
+    description = "Faster and more efficient replacement for std::unordered_map / std::unordered_set"
+    topics = ("conan", "robin-hood-hashing", "header-only", "containers")
+    homepage = "https://github.com/martinus/robin-hood-hashing"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+
```
Comment (madebr) [inline recipes/robin-hood-hashing/all/conanfile.py:14]:
```suggestion
    license = "MIT"
    settings = "compiler"
```

File: recipes/robin-hood-hashing/all/conanfile.py (hunk 8 above)
Comment (Talkless) [reply recipes/robin-hood-hashing/all/conanfile.py:24]: It looks it supports down to 11.

File: recipes/robin-hood-hashing/all/conanfile.py (hunk 9 above)
Comment (Talkless) [reply recipes/robin-hood-hashing/all/conanfile.py:14]: Why do we need compiler? Conan docs says we can skip `settings` for header-only..?

Comment (Talkless) [reply recipes/robin-hood-hashing/all/conanfile.py:14]: Oh it's for C++ standard check...

File: recipes/robin-hood-hashing/all/test_package/CMakeLists.txt (hunk 5 above)
Comment (Talkless) [reply recipes/robin-hood-hashing/all/test_package/CMakeLists.txt:12]: It does not find `robin_hood.h` if I change `CONAN_PKG::robin_hood` into `${CONAN_LIBS}`.

Comment (madebr) [reply recipes/robin-hood-hashing/all/test_package/CMakeLists.txt]: Ah yes, you're doing ´conan_basic_setup(TARGETS)´, which does not do ´include_directories´

File: recipes/robin-hood-hashing/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+class ConanRecipe(ConanFile):
+    name = "robin-hood-hashing"
+    description = "Faster and more efficient replacement for std::unordered_map / std::unordered_set"
+    topics = ("conan", "robin-hood-hashing", "header-only", "containers")
+    homepage = "https://github.com/martinus/robin-hood-hashing"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "compiler"
+
+    generators = "cmake"
```
Comment (prince-chrismc) [inline recipes/robin-hood-hashing/all/conanfile.py:14]: There does not seem to be any deps to this project
```suggestion
```

File: recipes/robin-hood-hashing/all/test_package/CMakeLists.txt (hunk 5 above)
Comment (Talkless) [reply recipes/robin-hood-hashing/all/test_package/CMakeLists.txt]: I copied test_package form fmt I believe, or cach2, don't really remember...

Comment (Talkless) [reply recipes/robin-hood-hashing/all/test_package/CMakeLists.txt]: Fixed, with `${CONAN_LIBS}` now.

**Comment by Talkless** [comment]
Thanks @madebr and @prince-chrismc for all comments, leaned a lot!

**Pull request merged**